### PR TITLE
Create Python/hamming/mentoring.md

### PR DESCRIPTION
**Pull request by jeffdparker**
A first draft of notes for mentoring hamming on the Python track

### CODE REVIEW

**Comment by rpottsoh** [comment]
Congrats Jeff on getting the PR created properly.

File: tracks/python/exercises/hamming/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+The solution needs to raise an exception if the lengths are different
+
+Otherwise, the task is to find the number of differences between two strings of the same size.
+
+### Reasonable solutions
+	
+We check the lengths, and then look for differences
+	
+    ```python
```
Comment (SleeplessByte) [inline tracks/python/exercises/hamming/mentoring.md]: Remove indentation

File: tracks/python/exercises/hamming/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return count
+    ```
+
+This works, but needs to use an index variable to walk the two strings.
+
+#### Introducing zip()
+
+We can use the zip function to pair up the two strands
+
+    ```python
```
Comment (SleeplessByte) [inline tracks/python/exercises/hamming/mentoring.md]: Remove indentation

File: tracks/python/exercises/hamming/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                count = count + 1
+
+        return count
+    ```
+    
+#### List Comprehensions
+	
+The solution below uses a List Comprehension to provide a more terse approach:
+	
+    ```python
```
Comment (SleeplessByte) [inline tracks/python/exercises/hamming/mentoring.md]: Remove indentation

File: tracks/python/exercises/hamming/mentoring.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        return len([(a, b) for (a, b) in zip(strand_a, strand_b) if a != b])
+    ```
+	
+### Common Suggestions
+
+A List Comprehension can be suggested, but isn't an obvious win.
+	 
+### Talking points
+
+Running time is not a concern here.  It is hard to make this worse than O(N).
```
Comment (SleeplessByte) [inline tracks/python/exercises/hamming/mentoring.md]: Double space

Comment (jeffdparker) [reply tracks/python/exercises/hamming/mentoring.md]: Sorry - double space where?

Comment (NobbZ) [reply tracks/python/exercises/hamming/mentoring.md]: After the for. I do not consider this a bad thing though.

Comment (SleeplessByte) [reply tracks/python/exercises/hamming/mentoring.md]: After the dot, 

```
Running time is not a concern here.__It is hard to make this worse than O(N).
```

Comment (SleeplessByte) [reply tracks/python/exercises/hamming/mentoring.md]: > After the dot. I do not consider this a bad thing though.
It's not consistent with the other mentor notes. Either have it everywhere or nowhere.

Comment (NobbZ) [reply tracks/python/exercises/hamming/mentoring.md]: As an part-time emacs user I'd suggest to add it everywhere else ;)

But I do understand what you mean. Yes, for beeing consistent it should be removed.

**Comment by SleeplessByte** [comment]
Hehe :). I just always rely on auto format and even GitHub (or my chrome?)
Edited it away the first time I copied the sentence.

**Pull request merged**